### PR TITLE
Em/emip

### PR DESCRIPTION
**Pull request by sgkang**
Mix of working on spectral IP + MT tutorial 

Updates in IP coes
- Store sensitivity matrix 

Updates in directives (for MT tutorial)
- Update `SaveOutputEveryIteration`
- Add few plotting utilities for that

### CODE REVIEW

File: SimPEG/EM/Static/SIP/Regularization.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from SimPEG import Utils, Maps, Mesh
+import numpy as np
+import scipy.sparse as sp
+from SimPEG.Regularization import BaseRegularization, Simple
+
+
+class MultiRegularization(BaseRegularization):
```
Comment (lheagy) [inline SimPEG/EM/Static/SIP/Regularization.py]: Do we need to add this back? or can this be replaced by the new objective function implementation?

**Comment by lheagy** [comment]
we should get #614 in first as the TDEM changes are there already

File: SimPEG/EM/Static/SIP/Regularization.py (hunk 1 above)
Comment (lheagy) [reply SimPEG/EM/Static/SIP/Regularization.py]: @sgkang: what are your thoughts on this?

**Comment by lheagy** [comment]
hi @sgkang: are these still changes you would like to bring in?

**Comment by sgkang** [comment]
Yes. I do want to bring in.

File: SimPEG/EM/Static/SIP/Regularization.py (hunk 1 above)
Comment (sgkang) [reply SimPEG/EM/Static/SIP/Regularization.py]: I'll deprecate this.

**Comment by lheagy** [comment]
same question here - looks like quite a few merge conflicts?

**Comment by lheagy** [comment]
@sgkang: it looks like there are quite a few merge conflicts here, is this still a live pr, or have you merged in changes elsewhere?

**Comment by sgkang** [comment]
connected to #715

**Pull request closed without merging**